src/vs/workbench/parts/terminal/electron-browser/terminalService.ts

```diff
@@ -230,6 +230,9 @@ export class TerminalService implements ITerminalService {
 			env[`PTYSHELLARG${i}`] = arg;
 		});
 		env['PTYCWD'] = this.contextService.getWorkspace() ? this.contextService.getWorkspace().resource.fsPath : os.homedir();
+		if (!env['LANG'] && platform.locale) {
```

Comment:
This function's a bit busy at this point, can you factor out the `env` stuff into a `createTerminalEnv` function?